lib/telegram_step.rb

```diff
@@ -1,26 +1,37 @@
+require 'faraday'
 require 'telegram/bot'
 
 class TelegramStep
+  attr_writer :conn
+
   def initialize(token, chat_id, payload)
     @token = token
     @chat_id = chat_id
     @payload = payload
   end
 
+  def conn
```

Comment:
Not entirely sure if `conn` is needed here. I know this will save you some lines of code on `dollar_step` and `weather_step` but it adds unnecessary baggage to subclasses that don't need to make API requests.

Reply:
Yep... I've added it just to test easier and simplify the constructors, and forget to check it thoroughly. I think that adding a connection object I could address both issues.

Reply:
After having a look, I think that there shouldn't be a problem with `conn`. It's the same as `send_message` and `send_gif` methods, which are not required for all subclasses. And `Faraday` object will not be created until someone calls the `conn` method.